src/features/locators/components/LocatorEditDialog.tsx

```diff
@@ -1,76 +1,98 @@
 import FormItem from "antd/es/form/FormItem";
-import { LocatorType } from "../../../common/types/locatorType";
+import { LocatorType, SelectOption } from "../../../common/types/common";
 import { getLocator, getXPathByPriority } from "../utils/locatorOutput";
+import { sendMessage } from "../../../pageServices/connector";
+import { evaluateXpath } from "../utils/utils";
+import { generateId, getElementFullXpath, isFilteredSelect } from "../../../common/utils/helpers";
+import { newLocatorStub } from "../utils/constants";
 
 interface Props extends Locator {
   isModalOpen: boolean;
   setIsModalOpen: (value: boolean) => void;
-  library: ElementLibrary;
+  isCreatingForm?: boolean;
 }
 
-interface EditFormProps {
+interface FormValues {
   name: string;
   type: ElementClass;
+  locatorType: LocatorType;
   locator: string;
 }
 
 export const LocatorEditDialog: React.FC<Props> = ({
   isModalOpen,
   setIsModalOpen,
+  isCreatingForm = false,
   element_id,
-  isCustomName,
+  isCustomName = true,
   name,
   type,
   locator,
-  library,
   jdnHash,
   validity,
   elemId,
   elemName,
   elemText,
   locatorType,
 }) => {
-  const isValidationEnabled = useLocatorValidationEnabled();
-  const [locatorValidity, setLocatorValidity] = useState<string>(isValidationEnabled ? validity?.locator || "" : "");
-  const [isEditedName, setIsEditedName] = useState<boolean>(Boolean(isCustomName));
-
+  const dispatch = useDispatch();
   const locators = useSelector(selectLocatorsByPageObject);
   const types = useSelector((_state: RootState) => selectAvailableClasses(_state));
   const pageObjectLocatorType = useSelector(selectCurrentPageObject)?.locatorType;
+  const pageObjectId = useSelector(selectCurrentPageObject)!.id;
+  const library = useSelector(selectCurrentPageObject)?.library || defaultLibrary;
 
-  const [formLocatorType, setLocatorType] = useState<LocatorType>(
-    locatorType || pageObjectLocatorType || LocatorType.xPath
+  const isValidationEnabled = useLocatorValidationEnabled();
+  const [{ message, validationStatus }, setLocatorValidity] = useState<Validity>(
+    isValidationEnabled
+      ? { message: validity?.message || "", validationStatus: validity?.validationStatus || "" }
+      : { message: "", validationStatus: ValidationStatus.WARNING }
   );
 
-  const [form] = Form.useForm<EditFormProps>();
-  const dispatch = useDispatch();
+  const [isEditedName, setIsEditedName] = useState<boolean>(isCustomName);
+
+  // should be reduced when we'll enable css locators creating
+  const getFormLocatorType = () =>
+    isCreatingForm ? LocatorType.xPath : locatorType || pageObjectLocatorType || LocatorType.xPath;
+
+  const [formLocatorType, setLocatorType] = useState<LocatorType>(getFormLocatorType());
+  const [form] = Form.useForm<FormValues>();
+  const initialValues: FormValues = {
+    type,
+    name: name || "",
+    locator: locator ? getXPathByPriority(locator) : "",
```

Comment:
why you don't take `locator.output` here?

Reply:
looks like a logic that I forgot to update with ccs-selector feature

Reply:
untill we have only xpath for locatorType as a initial value, it will work. you can leave it as is for now

Reply:
if we have css-selector as a locatorType, here will be an error

Reply:
fixed